File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/Source/DKImagePickerController/View/DKAssetGroupGridLayout.m
//
//  DKAssetGroupGridLayout.m
//  DKImagePickerControllerDemo_OC
//
//  Created by zm on 2017/6/26.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "DKAssetGroupGridLayout.h"

@implementation DKAssetGroupGridLayout
- (void)prepareLayout{
    [super prepareLayout];
    CGFloat minItemWidth = 80.0;
    if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
        minItemWidth = 100.0;
    }
    
    CGFloat interval  = 1.0;
    self.minimumInteritemSpacing = interval;
    self.minimumLineSpacing = interval;
    
    CGFloat contentWidth = self.collectionView.bounds.size.width;
    NSInteger itemCount =  floor(contentWidth / minItemWidth);
    CGFloat itemWidth = (contentWidth - (interval * (itemCount - 1))) / itemCount;
    CGFloat actualInterval =( contentWidth - itemCount * itemWidth ) / (itemCount - 1);
    
    itemWidth += actualInterval - interval;
    
    self.itemSize = CGSizeMake(itemWidth, itemWidth);
    
}
@end

File: DKImagePickerController/View/DKAssetGroupGridLayout.swift
//
//  DKAssetGroupGridLayout.swift
//  DKImagePickerControllerDemo
//
//  Created by ZhangAo on 16/1/17.
//  Copyright © 2016年 ZhangAo. All rights reserved.
//

import UIKit

open class DKAssetGroupGridLayout: UICollectionViewFlowLayout {
	
	open override func prepare() {
		super.prepare()
		
		var minItemWidth: CGFloat = 80
		if UI_USER_INTERFACE_IDIOM() == .pad {
			minItemWidth = 100
		}
		
		let interval: CGFloat = 1
		self.minimumInteritemSpacing = interval
		self.minimumLineSpacing = interval
		
		let contentWidth = self.collectionView!.bounds.width
		
		let itemCount = Int(floor(contentWidth / minItemWidth))
		var itemWidth = (contentWidth - interval * (CGFloat(itemCount) - 1)) / CGFloat(itemCount)
		let actualInterval = (contentWidth - CGFloat(itemCount) * itemWidth) / (CGFloat(itemCount) - 1)
		itemWidth += actualInterval - interval
		
		let itemSize = CGSize(width: itemWidth, height: itemWidth)
		self.itemSize = itemSize
	}
	
}
